src/Mynt.Core/Backtester/DataRefresher.cs

```diff
@@ -3,52 +3,53 @@
 using System.IO;
 using System.Linq;
 using System.Threading.Tasks;
+using ExchangeSharp;
 using LiteDB;
+using Microsoft.Extensions.Configuration;
 using Mynt.Core.Enums;
 using Mynt.Core.Exchanges;
 using Mynt.Core.Extensions;
 using Mynt.Core.Models;
 using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace Mynt.Core.Backtester
 {
     public class DataRefresher
     {
-        private readonly ExchangeOptions _exchangeOptions;
-        private readonly BaseExchange _api;
-
-        public DataRefresher(ExchangeOptions exchangeOptions)
+        public static bool CheckForCandleData()
         {
-            _exchangeOptions = exchangeOptions;
-            _api = new BaseExchange(_exchangeOptions);
+            return Directory.GetFiles(BacktesterDatabase.GetDataDirectory(), "*.db", SearchOption.TopDirectoryOnly).Count() != 0;
         }
 
-        public bool CheckForCandleData()
+        public static async Task RefreshCandleData(Action<string> callback, BacktestOptions backtestOptions)
         {
-            return Directory.GetFiles(BacktesterDatabase.GetDataDirectory(), "*.db", SearchOption.TopDirectoryOnly).Count() != 0;
-        }
 
+            var builder = new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile("appsettings.json", optional: true);
```

Comment:
The main reason the abstraction is in place is to be able to unplug ExchangeSharp from the project and move in something else if an alternative arises without having to change the entire codebase. Currently all actual ExchangeSharp code is contained to the `BaseExchange` which is the only point we need to adjust if it's decided to use something else.

Adding new exchanges is as simple as implementing a new enum value and hooking it up in the BaseExchange constructor. Also, not all exchanges work the same way (e.g. Huobi and OKEx) so some additional code is required in some cases. That can all be arranged in the `BaseExchange` so I'm not too enthousiastic about moving away from that.

I'm also not entirely sure what you mean with this comment? What exactly isn't a good solution? And also, since we're only using candle data here that is (in all supported exchanges so far) publicly available without API keys etc.

Reply:
This (my commit) solution is not good because it's still use the default exchange from config and could not be overwritten from a optional Exchange in e.g. a API -> I've changed this in last commit :-)